geonode/resource/manager.py

```diff
@@ -447,23 +447,21 @@ def copy(self, instance: ResourceBase, /, uuid: str = None, owner: settings.AUTH
                     _resource = copy.copy(instance.get_real_instance())
                     _resource.pk = _resource.id = None
                     _resource.uuid = uuid or str(uuid4())
+                    _resource.save()
```

Comment:
We also get an error on saving the dataset `DETAIL:  Key (store, workspace, name)=(geonode_data, geonode, {name}) already exists.`

Reply:
Thanks @marthamareal pushed a fix and test case for this one